### PR TITLE
MAINT/BENCH: sparse: Benchmark cleanup and additions

### PR DESCRIPTION
**Pull request by perimosocordiae**
It seemed like these benchmarks were either not doing what we wanted, or were doing too much work in the setup phase.

@pv, if I've somehow misunderstood the benchmarking system, please let me know.

### CODE REVIEW

**Comment by pv** [comment]
It's mostly copypaste-conversion from the old benchmarking code, so that's why it looks weird.

I'd suggest avoiding changing the `params` of the benchmarks unless necessary, as this invalidates existing benchmark results. Of course they can be regenerated, but best to be sure it's necessary.

**Comment by perimosocordiae** [comment]
Ok, I'll roll back some of my more aggressive changes to keep `params` consistent.

What about the case of `Matmul`, which didn't have any params listed?

**Comment by pv** [comment]
With Matmul it's the same --- changing params will invalidate old results. Of course, if the change seems necessary, go ahead.

File: benchmarks/benchmarks/sparse.py [hunk 1]
```diff
@@ -79,92 +75,72 @@ def time_arithmetic(self, format, XY, op):
...
-        A = self.matrices['dia']
-        self.x = ones((A.shape[1], 10), dtype=A.dtype)
+    def setup(self, n, format):
+        self.A = poisson2d(n, format=format)
+        self.x = ones((self.A.shape[1], 10), dtype=self.A.dtype)
 
-    def time_matvecs(self, fmt):
-        A = self.matrices[fmt]
-        A*self.x
+    def time_matvecs(self, n,  format):
```
Comment (larsmans) [inline benchmarks/benchmarks/sparse.py]: CI failure is caused by the pair of spaces before `format`.

**Comment by perimosocordiae** [comment]
@pv: I've reverted some of the unnecessary changes to the `params` lists, though I left `Matvec` changed as I think it'll be useful to test a bigger matrix of sparse matrix types and formats.

File: benchmarks/benchmarks/sparse.py [hunk 2]
```diff
@@ -83,64 +79,52 @@ class Sort(Benchmark):
...
-        K = int(float(K))
-        self.A = random_sparse(N, N, K)
+        n = 10000
+        if matrix.startswith('Rand'):
+            k = int(matrix[4:])
+            self.A = random_sparse(n, n, k)
+            self.A.has_sorted_indices = False
+            self.A.indices[:2] = 2, 1
+        else:
+            raise NotImplementedError()
```
Comment (pv) [inline benchmarks/benchmarks/sparse.py]: This is equivalent to skip --- maybe better to raise RuntimeError

**Pull request merged**